lib/account.js

```diff
@@ -2,269 +2,295 @@ var Boom = require('boom');
 var hyperquest = require('hyperquest');
 var url = require('url');
 
-module.exports = function(config) {
-  // https://basic:auth@login.server.org
-  var loginAPI = config.loginAPI;
-
-  function getIPAddress(request) {
-    // account for load balancer!
-    if (request.headers['x-forwarded-for']) {
-      return request.headers['x-forwarded-for'];
-    }
-
-    return request.info.remoteAddress;
+function getIPAddress(request) {
+  // account for load balancer!
+  if (request.headers['x-forwarded-for']) {
+    return request.headers['x-forwarded-for'];
   }
 
-  function parseMessage(message, callback) {
-    var bodyParts = [];
-    var bytes = 0;
+  return request.info.remoteAddress;
+}
 
-    message.on('data', function(c) {
-      bodyParts.push(c);
-      bytes += c.length;
-    });
+function parseMessage(message, callback) {
+  var bodyParts = [];
+  var bytes = 0;
 
-    message.on('end', function() {
-      var body = Buffer.concat(bodyParts, bytes).toString('utf8');
-      var json;
+  message.on('data', function(c) {
+    bodyParts.push(c);
+    bytes += c.length;
+  });
 
-      if ( message.statusCode !== 200 ) {
-        try {
-          json = JSON.parse(body);
-        } catch (ex) {
-          return callback(Boom.create(message.statusCode, 'LoginAPI error', body));
-        }
-        return callback(Boom.create(message.statusCode, 'LoginAPI error', json));
-      }
+  message.on('end', function() {
+    var body = Buffer.concat(bodyParts, bytes).toString('utf8');
+    var json;
 
+    if ( message.statusCode !== 200 ) {
       try {
         json = JSON.parse(body);
       } catch (ex) {
-        return callback(Boom.badImplementation('Error parsing response from Login server', ex));
+        return callback(Boom.create(message.statusCode, 'LoginAPI error', body));
       }
+      return callback(Boom.create(message.statusCode, 'LoginAPI error', json));
+    }
+
+    try {
```

Comment:
Why do this in two spots, since you seem to need the JSON parsed no matter the `statusCode`.  Why not do it above 27?